### ISSUE TITLE
[OJS] Improve UI/UX with names for multlang journals.

### ISSUE THREAD

**Issue by marcbria** [pkp/pkp-lib#6291]
Hi,

**Problem to solve:**
1. Avoid asking user names more than once when your journal asks for multiple languages... but with same charset. 
2. Improve the way you deal with names in multilang journals when you have different charsets.

** Proposal **
![UI-UX-mockup](https://user-images.githubusercontent.com/2779168/96442106-e243d000-120a-11eb-9570-b9a5885029e2.png)

- Field (1) is not a lang field, is a "generic" one.
- Field (2) is in Spanish.
- Field (3) is in Russian

When you write a text in Field (1), it autofills all the other fields behind it (2) and (3).
When you write in field (2) or (3), it only changes it's own content.
When you modify Field (1), it changes all the fields that match with it's own content (in the example, field (2)).

** Benefits **
This solution fixes problems issues commented in "Problems to solve"
- Problem 1: When you like same name in all langs (ie: same charset) you only need to fill field (1) and the rest of the langs will be autofilled.
- Problem 2: When you want different names in different langs (ie: different charset), you only need to fill field (1) and then modify the langs that change.

** External references **
Issue comes from this conversation in forum: https://forum.pkp.sfu.ca/t/configure-author-name-conventions/53738/11

**Comment by alvaroarmo** [comment]
I think this is a very interesting idea. I also find quite annoying the multiple language fields waiting to be filled when it comes to authors' names.

**Comment by NateWr** [comment]
It's my understanding that the given name should only be required in the primary language. The given name (and preferred name) fields should always fall back to the primary language, so that users should only ever need to enter it in one language. 

If this isn't the case then we should change the form to allow it. User registrations in the frontend, for example, do not require a name in more than one language.

**Comment by alexxxmendonca** [comment]
I fully support this.

**Comment by marcbria** [comment]
Yes Nate, now it works as you explain and this is much better than in first 3.1.x but the way it works is still a little confusing for our users.
Some issues (probably nuances, but make the user experience not full satisfactory) as I see it:
- The red icon make some users arrive to the conclusion they necessarily need to introduce their name in all the other name fields. Empty fields don't help to let them understand they are optional.
- If you go into one of those fields and introduce something, you won't be able to remove it because OJS will expect this lang to have some content (shows a "required field" error).

I'm not full happy with an interface that "does operations in user's behalf" (don't know how to express it better in English... in other words I would prefer a simpler solution) but with the proposal I did I feel like it's an average solution that can make everybody happy:
- Metadata will be complete: Green icon.
- You can operate multiple fields at the same time.
- You can keep different names in each lang.

Please @alexxxmendonca and @alvaroarmo extend the list of problems if you find I'm not describing the problem properly.

**Comment by NateWr** [comment]
I think that the UX issues will be addressed when we're able to convert these forms over to the new form system, where multilingual fields are less intrusive. To be frank, overriding the way the multilingual fields work for one form is going to be tricky to do and prone to breaking. I'd recommend that we prioritise moving the user entities over to the API so that we can use the new forms here. You'll get the best of both worlds.

If you need an improvment more quickly, a simple plugin could be written to override this form template with your own.

**Comment by marcbria** [comment]
Is there any example of how the new multiligual fields will work?
I mean, in codepen.io or jsfiddle.net to have an idea...
At the end, now, this is not a huge problem... so it will be probably addressed as you said.

**Comment by NateWr** [comment]
The new forms can be seen in 3.2+ in the settings forms and the publication tab of the workflow. You can also explore all of the field components in our [UI Library](https://docs.pkp.sfu.ca/dev/ui-library/dev/#/component/Form). Eventually all of our forms will follow this new pattern.

**Comment by sbirAECA** [comment]
> The new forms can be seen in 3.2+ in the settings forms and the publication tab of the workflow. You can also explore all of the field components in our [UI Library](https://docs.pkp.sfu.ca/dev/ui-library/dev/#/component/Form). Eventually all of our forms will follow this new pattern.

Thank you @NateWr . Do you have any idea when all the forms will be under the new pattern?. I know it will make customization more flexible, few days ago I sent this to the pkp forum ([OJS 3.2.1.1 Need to change field type from FieldRichTextarea to FieldTextarea in e.g Masthead form](https://forum.pkp.sfu.ca/t/ojs-3-2-1-1-need-to-change-field-type-from-fieldrichtextarea-to-fieldtextarea-in-e-g-masthead-form/63509)). Not having to deal with template files is a relief.
But, for all those users who do not have development capacity, can there be a user interface to customize, not only the properties and characteristics of existing fields, but also the possibility of adding or removing fields?

**Comment by NateWr** [comment]
> But, for all those users who do not have development capacity, can there be a user interface to customize, not only the properties and characteristics of existing fields, but also the possibility of adding or removing fields?

We'll probably never build this because it would be a lot of work to create and maintain. And for most use cases adding properties and fields isn't sufficient. It's also important how that data is mapped to published material, metadata deposits, etc. So there is still a coding requirement to do something with it.

But we hope in the future it may make customisability easier for certain use cases. For example, the theme options API makes it easy to add theme options in your custom theme. We'll eventually use these components to improve the custom review forms, and we may implement settings to enable/disable fields in the submission form.

**Comment by marcbria** [comment]
> You can also explore all of the field components in our UI Library.

I forget it, but the UI library is great. Thanks for the work.

As you said, they are less intrusive, but I have doubts if will address the 2 issues I pointed in former post:

1. **Red icon:** The red icon shown if you don't fill all the multilang variants will make some users arrive to the conclusion they necessarily need to introduce their name in all the other name fields.
2. **Required field:** If you go into one of those fields and introduce something, you won't be able to remove it because OJS will expect this lang to have some content (shows a "required field" error).

I suspect you have 2 in your toDo list, but I'm curious about 1... Any idea about how to fix this?

**Comment by sbirAECA** [comment]
Good morning. I started testing OJS 3.2.1.1, and I can see that context, emailTemplate, publication and site were under the new pattern. In the github repository has been added FORM_ANNOUNCEMENT. Can you tell me if user and submission will be under the new pattern soon? 
Thank you

**Comment by NateWr** [comment]
> I suspect you have 2 in your toDo list

Once moved over to the new API, this problem will go away. I think it's related to how required fields are applied in the old system, and how that interacts with a third-party validation library that is running on the old forms.

Let's wait and see on the red icon. If it continues to be a common problem we can address it then.

> Can you tell me if user and submission will be under the new pattern soon?

We are working to update the submission form. As a first step, 3.3 will introduce a new submission file upload component (see #6057). Then in 3.4 we are planning to migrate the whole submission form over. This is part of our accessibility work planned in the [roadmap](https://pkp.sfu.ca/roadmap).

We don't have any work scheduled that would migrate the user forms, but I hope that we will find an opportunity to do this soon. In particular, we are constantly getting requests to make it easier to modify the author forms (which are closely connected to the user forms). :crossed_fingers:

**Comment by marcbria** [comment]
Let's see how it looks like in 3.3 but, as a brainstorm proposal... what about showing that some multilang fields are optional?
Imagine a world icon in black (or gray, even it also expresses some kind of incompleteness), till you click on it and you fill a variant (that will change to red unless you fill all the variants).

In this way we avoid the confusing red-world-icon that makes some users think they need to complete all fields to make it green.

**Comment by NateWr** [comment]
Isn't this how they work now? In [this example](https://docs.pkp.sfu.ca/dev/ui-library/dev/#/component/Form/ExampleMultilingual), the affiliation field is grey until at least one value is entered.

**Comment by marcbria** [comment]
Conversation derivates to UI/UX stuff but the original issue we were trying to explain is still there.

**In short:** Whey you have multilang, but all langs are in same charset (a very common scenario)... don't make any sense to ask for user name/surname for each lang. Existing UI/UX is confusing for final users that think they need to fill the name/surname fields for every lang.

Explained better here:
https://forum.pkp.sfu.ca/t/changing-default-language-impacts-on-users-given-family-names/67739/3

The "feature request" is keeping all langs sync, but invisible to final user (not exposed in the UI)... unless user asks for "unsync" (replace the world icon by a "unsync" icon?).

Please @NateWr do you mind to move this to "Feature request" too.

**Comment by Michevole** [comment]
> 
> The "feature request" is keeping all langs sync, but invisible to final user (not exposed in the UI)... unless user asks for "unsync" (replace the world icon by a "unsync" icon?).
> 

I think this improvement is really indispensable for those sites who host many journals and have many editors and countless authors. The journal manager's work is costantly interrupted by users asking for support for an unnecessary field that stop them.

It's also difficult for me to explain why their journals need that double item. I absolutely support this request!

**Comment by marcbria** [comment]
@NateWr is this clear enough, or do you think it is better to take a detailed look in Discussion before starting coding a single line?

I mean, no hurry at all (I know you all are overwhelmed), but I don't know if the solution you have in mind will fix the 3 problems I mention at the beginning and specifically the most important one that is "there is NO need to ask user's names/surnames more than once" when your journal only allow langs with same charset.
